### PR TITLE
[fixes #16] allow to pass dynamic offsets

### LINKED ISSUE #16: offsets is wrong when using with pino-debug
i'm using pino-debug along with pino-caller, changing the stack trace which is then printing the wrong line: 
```Error
    at eval (eval at asJson (/project/node_modules/pino-caller/index.js:16:5), <anonymous>:1:7)
    at Object.asJson (/project/node_modules/pino-caller/index.js:16:5)
    at Object.write (/project/node_modules/pino/lib/proto.js:140:28)
    at Object.LOG [as debug] (/project/node_modules/pino/lib/tools.js:37:26)
    at enabled (/project/node_modules/pino-debug/debug.js:19:16)
    at Object.<anonymous> (/project/packages/something/lib/index.js:12:16)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)```

will print `"Object.LOG [as debug] (/project/node_modules/pino/lib/tools.js:37:26)"`

It would be nice to be able to tweak the offsets.

### PR DESCRIPTION
**Pull request by gregory**

### CODE REVIEW

**Comment by mcollina** [comment]
Would you mind:

* [ ] add package-lock.json to .gitignore
* [ ] remove package-lock from this PR
* [ ] add Node 12 to .travis.yml
* [ ] add the new option to the README

Thanks!

File: index.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 const NODEJS_VERSION = parseInt(process.version.slice(1).split('.')[0], 10)
-const STACKTRACE_OFFSET = NODEJS_VERSION && NODEJS_VERSION > 6 ? 4 : 5
-const LINE_OFFSET = 7
+const DEFAULT_STACKTRACE_OFFSET = NODEJS_VERSION && NODEJS_VERSION > 6 ? 4 : 5
+const DEFAULT_LINE_OFFSET = 7
 const { symbols } = require('pino')
 const { asJsonSym } = symbols
 
-function traceCaller (pinoInstance) {
+function traceCaller (pinoInstance, {stackTraceOffset = 0, lineOffset = 0} = {}) {
```
Comment (phra) [inline index.js]: imho no need to change the semantic of the function, see #18 for a more generic approach.

File: .travis.yml [hunk 2]
```diff
@@ -4,5 +4,6 @@ node_js:
 - '6'
 - '8'
 - '10'
+- '12'
```
Comment (phra) [inline .travis.yml]: would you mind to send a separate PR for updating the travis config?

File: tests/test.js [hunk 3]
```diff
@@ -24,7 +41,7 @@ test('pino caller works also when adding level', function (t) {
 
   const pinoInstance = pinoCaller(pino(through2(function (chunk, enc, callback) {
     const res = JSON.parse(chunk.toString('utf8'))
-    const regex = /Test.<anonymous> \(\/(.)*tests\/test.js/
+    const regex = /Proxy.fatal \(\/(.)*\/pino\/lib\/levels\.js/
```
Comment (phra) [inline tests/test.js]: `Test.<anonymous>` is the correct output of the test.

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 node_modules
+package-lock.json
```
Comment (phra) [inline .gitignore]: this one should be in a separated PR.

File: .travis.yml (hunk 2 above)
Comment (gregory) [reply .travis.yml]: done: https://github.com/pinojs/pino-caller/pull/19

File: .gitignore (hunk 4 above)
Comment (gregory) [reply .gitignore]: https://github.com/pinojs/pino-caller/pull/19

File: tests/test.js (hunk 3 above)
Comment (gregory) [reply tests/test.js]: not sure why but when i pulled the codebase the test was immediately failing because of that ...

Comment (gregory) [reply tests/test.js]: Same as Travis: https://travis-ci.org/pinojs/pino-caller/jobs/614735264?utm_medium=notification&utm_source=github_status

File: index.js (hunk 1 above)
Comment (gregory) [reply index.js]: This Pr allows the user to dynamically tweak based on the plugin or plugins they are going to use. 
#18 implies the bug is only happening because of `node_modules/pino`

File: tests/test.js (hunk 3 above)
Comment (gregory) [reply tests/test.js]: it seems that #18  is actually fixing that.

**Comment by phra** [comment]
superseded by #18
